services/channel/service/src/publishing/common/localization/get-validation-and-localizations.ts

```diff
@@ -0,0 +1,258 @@
+    );
+
+    let localizations: GqlLocalization[] | undefined = undefined;
+
+    if (data.validateEntityLocalizations.validationHash) {
+      const { data: publishData } =
+        await PrepareEntityLocalizationsForPublishing(
+          {
+            entityId,
+            entityType,
+            serviceId,
+            hash: data.validateEntityLocalizations.validationHash,
+          },
+          { Authorization: `Bearer ${authToken}` },
+        );
+      localizations =
+        publishData.prepareEntityLocalizationsForPublishing.localizations;
+    }
+
+    return {
+      localizations,
+      validation: data.validateEntityLocalizations,
+    };
+  } catch (e) {
+    throw getLocalizationMappedError(e);
+  }
+};
+
+export const getChannelValidationAndLocalizations = async (
+  localizationServiceBaseUrl: string,
+  authToken: string,
+  entityId: string,
+  entityType: string,
+  serviceId: string,
+): Promise<{
+  localizations: ChannelLocalization[] | undefined;
+  validations: PublishValidationMessage[];
+}> => {
+  try {
+    const { localizations, validation } = await getValidationAndLocalizations(
+      localizationServiceBaseUrl,
+      authToken,
+      entityId,
+      entityType,
+      serviceId,
+    );
+    return {
+      localizations: validateLocalizations(localizations)?.map((l) => ({
+        is_default_locale: l[LOCALIZATION_IS_DEFAULT_LOCALE],
+        language_tag: l[LOCALIZATION_LANGUAGE_TAG],
+        title: l.title,
+        description: l.description,
+      })),
+      validations: mapValidationMessages(
+        validation,
+        ({ locale, fieldName, message }: GqlValidationMessage) =>
+          `${locale}${fieldName ? `, ${fieldName}:` : ':'} ${message}`,
+      ),
+    };
+  } catch (e) {
+    throw getLocalizationMappedError(e);
+  }
+};
+
+export const getProgramValidationAndLocalizations = async (
+  localizationServiceBaseUrl: string,
+  authToken: string,
+  entityId: string,
+  entityType: string,
+  serviceId: string,
+  /**
+   * When playlist is published, we can get validation messages for multiple
+   * programs, so there is a need to identify which messages belong to which program.
+   */
+  programTitle?: string,
+): Promise<{
+  localizations?: ProgramLocalization[];
+  validations: PublishValidationMessage[];
+}> => {
+  try {
+    const { localizations, validation } = await getValidationAndLocalizations(
+      localizationServiceBaseUrl,
+      authToken,
+      entityId,
+      entityType,
+      serviceId,
+    );
+
+    return {
+      localizations: validateLocalizations(localizations)?.map((l) => ({
+        is_default_locale: l[LOCALIZATION_IS_DEFAULT_LOCALE],
+        language_tag: l[LOCALIZATION_LANGUAGE_TAG],
+        title: l.title,
+      })),
+      validations: mapValidationMessages(
+        validation,
+        ({ message }: GqlValidationMessage) =>
+          `${
+            programTitle
+              ? `The program "${programTitle}" is not fully localized: `
```

Comment:
This case bugs me just a little bit:
- Assign movie to the playlist
- Save by going back and publish the playlist
- Observe errors like `The program "Movie Title" is not fully localized: No localization value and status was found for the field "title" with the locale "bg-BG"`
- Wait for 10-15 seconds, refresh validation messages station, and observe that
  the errors are gone.

I understand that it might take some time because of the nature of how program
items are localized, and I'm not sure if there is a better way to handle it, but
in the current state, I would prefer to give the user an indication that
`localizations might not be synchronised yet, try refreshing in a couple of seconds`?